src/wallet/wallet.cpp

```diff
@@ -1772,7 +1773,7 @@ void CWalletTx::GetAmounts(std::list<COutputEntry>& listReceived,
         COutputEntry output = {address, txout.nValue, (int)i};
 
         // If we are debited by the transaction, add the output as a "sent" entry
-        if (nDebit > 0)
+        if (fIsAllFromMe)
```

Comment:
Does this mean that a transaction where some inputs are wallet related and none of the outputs are won't get registered as either 'sent' or 'received'?